### DISCUSSION TITLE
Black housing hides LEDs - what to do about it (if anything)?

### DISCUSSION THREAD

**Discussion by holger8080** [doudar/SmartSpin2k#180] [Q&A] [answered]
I'm building my device right now and a few more questions came up while doing so.

I just saw that in the code, the internal ESP32 LEDs are used. I printed the housing now in black PETG, so I won't be able to see the board LEDs. Are the LEDs needed and/or really useful? 

A few options come to mind:
- add external LEDs, reconfigure the code, "Frankenstein" wires onto pins of the ESP32 (I don't see any external LED pins on the board)
- drill a hole in the housing where the internal LEDs are positioned on the ESP32
- use waveguides

Before I do any of that: is it worth it? Now I know why you printed the housing in transparent PETG ;-) 

PS: great project, took an FTP test yesterday and really wished I had ERG-mode already.

**Comment by fireyeti** [comment, marked as answer]
The red LED indicates power to the ESP. The blue LED has a few functions. It is flashing at 1/sec when it is ready to pair to Bluetooth servers. One connected it is on steady. If the ESP32 firmware is updating, the blue LED flashes rapidly (once for each packet received).

I think it would work if drilled holes directly over the LEDs and placed a dab of hot glue over the hole as a transparent window for the LEDs.

**Reply by holger8080** [reply]
Cool, I'll try that.

**Reply by doudar** [reply]
Probably the biggest function of the leds is to determine if the unit is on or doing a firmware update. The slow flashing vs steady on of the blue LED when a client is connected is nice but not essential IMO.

**Comment by kadaan** [comment]
Mind posting a picture of your print?
